libvast/include/vast/detail/padded_buffer.hpp

```diff
@@ -0,0 +1,71 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <algorithm>
+#include <memory>
+#include <string_view>
+
+namespace vast::detail {
+
+template <auto PaddingSize, auto PaddingValue>
```

Comment:
I think this is easier to read.

```suggestion
template <size_t PaddingSize, char PaddingValue>
```

Reply:
Wanted to be more generic, but since the other class inernals are not generic then it may make sense to only allow size_t and char for now